### DISCUSSION TITLE
Does this look right?

### DISCUSSION THREAD

**Discussion by ShaneHash** [KarelSvo/ddt4all-5.6.0#2] [General]
I'm struggling to get my laptop to talk to my Twingo 3.

I have Pyren and DTT4All on a pretty old Windows 7 machine (T1350@1.86 with 4GB of memory) - it's years old.

I have a Forscan USB ODB2 box (pictured).

I have a modified Y-cable and there's an image here showing how the modification has been made.

But I'm not getting any success. 

![IMG_20241125_152345181](https://github.com/user-attachments/assets/0a02f3ae-4a49-448f-abc3-79f777aced47)

![IMG_20241125_152354932](https://github.com/user-attachments/assets/8164d105-ee0c-4dd9-b9df-45c7b0515256)

![IMG_20241206_221546316](https://github.com/user-attachments/assets/32265785-edf9-434d-93ec-d4a123db991f)

![odb2v2jpg](https://github.com/user-attachments/assets/d7bfbe97-728b-40b3-971a-22dcc543e57b)

Any ideas?

There's a log file here: 

[elm_ddt.txt](https://github.com/user-attachments/files/18044387/elm_ddt.txt)

**Comment by KarelSvo** [comment]
You cannot see in the log file whether the interface is working in general.
So scan in switch position HS without modifications.

**Reply by ShaneHash** [reply]
Here are logs for Pyren.

[elm_ddt (2).txt](https://github.com/user-attachments/files/18137858/elm_ddt.2.txt)

[ecu_ddt.txt](https://github.com/user-attachments/files/18137859/ecu_ddt.txt)

And here are logs for DDT4All.

[elm_log.txt](https://github.com/user-attachments/files/18137865/elm_log.txt)

[ecu_log.txt](https://github.com/user-attachments/files/18137866/ecu_log.txt)

[1734183093_TCU_GEN2_V3.2_20151109T114222.txt](https://github.com/user-attachments/files/18137869/1734183093_TCU_GEN2_V3.2_20151109T114222.txt)

[1734182905_TCU_GEN2_V3.2_20151109T114222.txt](https://github.com/user-attachments/files/18137873/1734182905_TCU_GEN2_V3.2_20151109T114222.txt)

I am at least seeing some communication with the ODB2 box attached directly to the ODB2 port in the car. 

I'm not sure how many ECUs I should be seeing, but because it is a Fordscan ODB2 I guess I should expect to see the media ECUs until I have sorted out the connections.

![Screenshot_1 (2)](https://github.com/user-attachments/assets/51f86616-b9f3-4d11-a62d-37d0286c8fe4)

I'm feeling a bit out of my depth, to be honest. It's going to take me months to get my head around this stuff... LOL

**Reply by KarelSvo** [reply]
Basically, your Elm interface works.
So you have to check your Can2 modification to make sure that the jumpers are not set in reverse.
With Pyren you scanned in CLIP mode. No ECUs were found because the CAN2 option was still activated on the start page.
ddt was not 5.6.0

**Reply by ShaneHash** [reply]
I'm very confused.

Just looked at the modified cable again and I can't work out if it is right or wrong.

With numbering system of the OBD2 port, the numbers go from 1 - 8 on the wider (above) side of the port and from 9 to 14 on the shorter (lower) side of the port. 

So, when doing the modification, you have to count the numbers from the back of the OBD2 plug, not the front of it. If you count from the front side of the cable plug, everything is reversed, (1 = 8, 9 =16 etc). 

So yes, I think I do have the modification back to front. 

It's very annoying...
